ext/libxml/config.w32

```diff
@@ -9,13 +9,19 @@ if (PHP_LIBXML == "yes") {
 			CHECK_HEADER_ADD_INCLUDE("libxml/tree.h", "CFLAGS_LIBXML", PHP_PHP_BUILD + "\\include\\libxml2") &&
 			ADD_EXTENSION_DEP('libxml', 'iconv')) {
 
-		EXTENSION("libxml", "libxml.c mime_sniff.c", false /* never shared */, "/DZEND_ENABLE_STATIC_TSRMLS_CACHE=1");
-		AC_DEFINE("HAVE_LIBXML", 1, "Define to 1 if the PHP extension 'libxml' is available.");
-		ADD_FLAG("CFLAGS_LIBXML", "/D LIBXML_STATIC /D LIBXML_STATIC_FOR_DLL /D HAVE_WIN32_THREADS ");
-		if (!PHP_LIBXML_SHARED) {
-			ADD_DEF_FILE("ext\\libxml\\php_libxml2.def");
+		if (GREP_HEADER("libxml/xmlversion.h", "#define\\s+LIBXML_VERSION\\s+(\\d+)", PHP_PHP_BUILD + "\\include\\libxml2") &&
```

Comment:
@petk @cmb69 This change assumes that xmlversion.h is in include/libxml2/libxml/xmlversion.h which is most likely not the case. The most likely place for xmlversion.h is include/libxml. With LIBXML_VERSION 20914 I got the warning that I did not have the required libxml version.
Dropping either the `\\libxml2` in the last argument of GREP_HEADER did the trick.

Reply:
I see. The GREP_HEADER function should be adjusted a bit, yes. Because the CHECK_HEADER_ADD_INCLUDE checks the include flags by appending them to the other flags, and this one works by a single file path only. It already checks the flags but only the empty PHP_EXTRA_INCLUDES variable (probably)...

Reply:
CHECK_HEADER_ADD_INCLUDE is smart enough to first check in the main include directory. With the headers in include/libxml it will find the headers. In my case:
>Checking for libxml/parser.h ...  ../win64build\include
>Checking for libxml/tree.h ...  ../win64build\include

for the x64 build

Reply:
`PHP_PHP_BUILD + "\\include\\libxml2"` in CHECK_HEADER_ADD_INCLUDE has been there for ages. I can see it there in PHP 7.0. In PHP 5.6 it simply was
> CHECK_HEADER_ADD_INCLUDE("libxml/parser.h", "CFLAGS_LIBXML")

Reply:
I'm testing something like this now:

```diff
diff --git a/win32/build/confutils.js b/win32/build/confutils.js
index 95a4e5ce3c..b652a497d1 100644
--- a/win32/build/confutils.js
+++ b/win32/build/confutils.js
@@ -965,6 +965,11 @@ function GREP_HEADER(header_name, regex, path_to_check)
 
        if (!c) {
                /* look in the include path */
+               if (path_to_check == null) {
+                       path_to_check = php_usual_include_suspects;
+               } else {
+                       path_to_check += ";" + php_usual_include_suspects;
+               }
 
                var p = search_paths(header_name, path_to_check, "INCLUDE");
                if (typeof(p) == "string") {
```

Reply:
> This change assumes that xmlversion.h is in include/libxml2/libxml/xmlversion.h which is most likely not the case. The most likely place for xmlversion.h is include/libxml. With LIBXML_VERSION 20914 I got the warning that I did not have the required libxml version.

From where did you get these libraries? I've checked a couple from windows.php.net, and even one from http://xmlsoft.org/sources/win32/64bit/, and there is always include/libxml2/libxml/*.h.

Reply:
I do not have my build machine at hand, but I probably just used https://github.com/winlibs/libxml2

Reply:
https://windows.php.net/downloads/php-sdk/deps/archives/vc11/x64/libxml2-2.9.4-1-vc11-x64.zip
And updates after that.

Reply:
Thanks for digging this out. So we need to fix this for best compatibility with other builds. I'll have a look at @petk's PR (thanks for that!) soon (probably the weekend).

Reply:
> Thanks for digging this out. So we need to fix this for best compatibility with other builds. I'll have a look at @petk's PR (thanks for that!) soon (probably the weekend).

@cmb69 Could you review the [PR](https://github.com/php/php-src/pull/15619) so that it can be merged?